clientd3d/d3drender.c

```diff
@@ -6866,8 +6867,11 @@ void D3DRenderObjectsDraw(d3d_render_pool_new *pPool, room_type *room,
 
 	anglePitch = PlayerGetHeightOffset();
 
-	// For each object rendered we increase the z-depth to prevent z-fighting.
-	int z_depth_inc = 0;
+	// We track all objects that are in similar positions in the 3D world.
+	// This is to mitigate z-fighting by incrementing z-depths for each unique object in the same position.
+	// The key is composed of the x and y coordinates of the object and the value is the current
+	// count of objects found at that location.
+	std::unordered_map<int64, BYTE> depth_adjustment_map;

```

Comment:
Let's make this an int instead of a BYTE.  I guess cast to BYTE later (I didn't realize z bias is only a BYTE... not sure why that is, maybe old video card limitations?).

Also pick int64 or int64_t consistently in this file.  I'd lean toward int64 for readability.